src/middleware/github-server-app-middleware.ts

```diff
@@ -44,9 +44,9 @@ export const GithubServerAppMiddleware = async (req: Request, res: Response, nex
 			uuid: gitHubServerApp.uuid,
 			hostname: gitHubServerApp.gitHubBaseUrl,
 			clientId: gitHubServerApp.gitHubClientId,
-			gitHubClientSecret: await gitHubServerApp.getDecryptedGitHubClientSecret(),
-			webhookSecret: await gitHubServerApp.getDecryptedWebhookSecret(),
-			privateKey: await gitHubServerApp.getDecryptedPrivateKey()
+			getDecryptedGitHubClientSecret: async () => gitHubServerApp.getDecryptedGitHubClientSecret(),
```

Comment:
Do we need `async` keyword here? `.getDecryptedBlahBlah()` returns a promise anyway, right?

Reply:
So ignoring the TS of method returning type, from JS engine point of view, I'd prefer explicit marking the function as async over implicit rely on the function impl return a Promise. Also marking it as `async` will allow the consumer to do `await`, instead of forcing them to do `xxx().then()`

----

But I could be wrong, just some theory from my js experience. Didn't verify what I said though lol :)

Reply:
Pretty sure they can await a sync function that returns a promise... But yeah, doesn't matter since we are removing this code altogether :nothingtodohere: